ehr_billing/api-src/org/labkey/api/ehr_billing/notification/BillingNotificationProvider.java

```diff
@@ -87,5 +88,20 @@ public interface BillingNotificationProvider
      * @return Name of center specific billing schema
      */
     public String getCenterSpecificBillingSchema();
-    
+
+    /**
+     * @param u User
+     * @param c Container where custom charge categories are defined
+     * @param startDate Filter custom charge category query where date >= startDate
+     * @param endDate Filter custom charge category query where date <= endDate
+     * @param totalsByCategory Add totals for the center-specific custom category to be displayed in the Charge Summary section of the notification.
+     * @param additionalCategoryUrls Add URLs for center-specific custom category to be linked to the Category
+     * See WNPRC Billing's implementation of this method for an example.
+     */
+    public default void getAdditionalChargeCategoryInfo(User u, Container c, Date startDate, Date endDate,
```

Comment:
It's hard to follow how the two maps are correlated (or not).

Assuming I'm following things correctly, I recommend making this a true `get` method, returning a `List<X>`, where a `X` has a String description, an ActionURL, a double total and a double totalCost.

Reply:
Agreed, its not clear. I've update the definition. I did not return List<X>, but instead return a Map and updated the code where its used, since that was easier to implement that way.

Reply:
This is much cleaner. But I'd still prefer to have a real Java class with the handful of supported properties instead of a map where we hard-code the property names as strings in multiple places.

Reply:
OK this is now cleaned up further